### PR TITLE
Use double instead of real_t type for time-related parameters and variables

### LINKED ISSUE #65313: _Process double type delta time introduces casting inconveniences
### Godot version

4.0.dev (b6d102c)

### System information

Fedora Linux 36

### Issue description

The change of node processing delta time to double, recently reflected in C# scripts, has surfaced some compilation errors that are fixable with new explicit downcasts to float:

AnimationTree.Advance(delta) : `delta` is float
Mathf methods don't accept parameters that are the result of arithmetical operations with double `delta` without downcasting

While this can be fixed with a few extra keywords I only wanted to have you consider this new situation.

### Steps to reproduce

```csharp
using Godot;

public partial class TestCase : Node3D
{
  public TestCase()
  {
    AddChild(animationTree);
  }

  public override void _Process(double delta)
  {
    var myVar = delta * 100;
    RotateObjectLocal(Vector3.Up, myVar); // error
    var myVar2 = Mathf.Cos(myVar); // error
    animationTree.Advance(delta); // error
  }

  private AnimationTree animationTree = new AnimationTree();
}
```

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by dpalais**
This is a draft pull request that addresses #65313 in that time-related variables and parameters types have been changed to use double type.
I changed everything I found except the Animation class and some Tween/Tweener types.
I left the Animation class alone as I ignore how much of its data is type delicate.
Tween/Tweener interpolators relies heavily on real_t values. I think it should be using double instead.

*Bugsquad edit:*
- Fixes #65313.

### CODE REVIEW

**Comment by fire** [comment]
Are you aware of the cicd github actions error. I support this work!

**Comment by dpalais** [comment]
I am aware, thanks for the heads up!

**Review by fire — APPROVED** [review]
Shape of this is great. @aaronfranke Can you check too?

**Review by aaronfranke — APPROVED** [review]
Did a quick look over the changes. Using `double` for time always makes sense. Assuming this works as expected, this is a good change.

File: scene/animation/tween.cpp [hunk 1]
```diff
@@ -71,16 +71,16 @@ void Tween::start_tweeners() {
...
-Ref<PropertyTweener> Tween::tween_property(Object *p_target, NodePath p_property, Variant p_to, float p_duration) {
+Ref<PropertyTweener> Tween::tween_property(Object *p_target, NodePath p_property, Variant p_to, double p_duration) {
 	ERR_FAIL_NULL_V(p_target, nullptr);
 	ERR_FAIL_COND_V_MSG(!valid, nullptr, "Tween invalid. Either finished or created outside scene tree.");
 	ERR_FAIL_COND_V_MSG(started, nullptr, "Can't append to a Tween that has started. Use stop() first.");
 
 	Variant::Type property_type = p_target->get_indexed(p_property.get_as_property_path().get_subnames()).get_type();
 	if (property_type != p_to.get_type()) {
-		// Cast p_to between floats and ints to avoid minor annoyances.
+		// Cast p_to between float64 and ints to avoid minor annoyances.
```
Comment (akien-mga) [inline scene/animation/tween.cpp:81]:
```suggestion
		// Cast p_to between double and int to avoid minor annoyances.
```

**Comment by akien-mga** [comment]
Looks good! Could you squash the commits? See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase) for instructions.

**Comment by dpalais** [comment]
@akien-mga done
thanks everyone

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**